### ISSUE TITLE
hwi 2.2.1 seem doesn't work on mac M1 chip

### ISSUE THREAD

**Issue by tongvanlinh** [bitcoin-core/HWI#676]
When I tried run hwi on my MAC M1, it throw error:

`./hwi enumerate
[2077] Module object for struct is NULL!
Traceback (most recent call last):
  File "struct.py", line 13, in <module>
ImportError: dlopen(/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so, 0x0002): tried: '/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so' (fcntl(fd, F_ADDFILESIGS_RETURN) failed with errno=7 in <FCF78D44-CB39-3D13-988F-D88C248A3D8C> '/private/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so', sliceOffset=0x00000000, codeBlobOffset=0x0000CD40, codeBlobSize=0x49545050), '/System/Volumes/Preboot/Cryptexes/OS/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so' (no such file), '/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so' (fcntl(fd, F_ADDFILESIGS_RETURN) failed with errno=7 in <FCF78D44-CB39-3D13-988F-D88C248A3D8C> '/private/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so', sliceOffset=0x00000000, codeBlobOffset=0x0000CD40, codeBlobSize=0x49545050), '/private/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so' (fcntl(fd, F_ADDFILESIGS_RETURN) failed with errno=7 in <FCF78D44-CB39-3D13-988F-D88C248A3D8C> '/private/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so', sliceOffset=0x00000000, codeBlobOffset=0x0000CD40, codeBlobSize=0x49545050), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so' (no such file), '/private/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so' (fcntl(fd, F_ADDFILESIGS_RETURN) failed with errno=7 in <FCF78D44-CB39-3D13-988F-D88C248A3D8C> '/private/var/folders/r6/4ttknn8d2y71k7d7_20n72fr0000gn/T/_MEI9camo9/lib-dynload/_struct.cpython-39-darwin.so', sliceOffset=0x00000000, codeBlobOffset=0x0000CD40, codeBlobSize=0x49545050)
[2077] Module object for pyimod01_archive is NULL!
Traceback (most recent call last):
  File "PyInstaller/loader/pyimod01_archive.py", line 27, in <module>
ModuleNotFoundError: No module named 'struct'
[2077] Module object for pyimod02_importers is NULL!
Traceback (most recent call last):
  File "PyInstaller/loader/pyimod02_importers.py", line 28, in <module>
ModuleNotFoundError: No module named 'pyimod01_archive'
Traceback (most recent call last):
  File "PyInstaller/loader/pyiboot01_bootstrap.py", line 17, in <module>
ModuleNotFoundError: No module named 'pyimod02_importers'
[2077] Failed to execute script 'pyiboot01_bootstrap' due to unhandled exception!
`
HWI version tested: https://github.com/bitcoin-core/HWI/releases/download/2.2.1/hwi-2.2.1-mac-amd64.tar.gz
MAC M1. OSX 11.3
Please support to resolve this issue.

**Comment by achow101** [comment]
M1 macs are currently unsupported. As indicated by the `amd64` in the filename, the only architecture supported by that binary is x86_64 (a.k.a. amd64).

**Comment by giahuy98** [comment]
This issue also occurs on Intel chip @achow101. The release binary hwi-2.2.1-mac-amd64 is not working, but hwi.py works.
https://github.com/nunchuk-io/nunchuk-desktop/issues/33#issuecomment-1513613209

```
~/Downloads/hwi-2.2.1-mac-amd64 ./hwi --debug enumerate
[38017] Module object for struct is NULL!
Traceback (most recent call last):
  File "struct.py", line 13, in <module>
ImportError: dlopen(/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so, 0x0002): tried: '/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so' (fcntl(fd, F_ADDFILESIGS_RETURN) failed with errno=7 in <FCF78D44-CB39-3D13-988F-D88C248A3D8C> '/private/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so', sliceOffset=0x00000000, codeBlobOffset=0x0000CD40, codeBlobSize=0x49545050), '/System/Volumes/Preboot/Cryptexes/OS/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so' (no such file), '/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so' (fcntl(fd, F_ADDFILESIGS_RETURN) failed with errno=7 in <FCF78D44-CB39-3D13-988F-D88C248A3D8C> '/private/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so', sliceOffset=0x00000000, codeBlobOffset=0x0000CD40, codeBlobSize=0x49545050), '/private/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so' (fcntl(fd, F_ADDFILESIGS_RETURN) failed with errno=7 in <FCF78D44-CB39-3D13-988F-D88C248A3D8C> '/private/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so', sliceOffset=0x00000000, codeBlobOffset=0x0000CD40, codeBlobSize=0x49545050), '/System/Volumes/Preboot/Cryptexes/OS/private/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so' (no such file), '/private/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so' (fcntl(fd, F_ADDFILESIGS_RETURN) failed with errno=7 in <FCF78D44-CB39-3D13-988F-D88C248A3D8C> '/private/var/folders/vg/gh08wlnd5rn3k_6drvzgdpc80000gn/T/_MEIZIJyih/lib-dynload/_struct.cpython-39-darwin.so', sliceOffset=0x00000000, codeBlobOffset=0x0000CD40, codeBlobSize=0x49545050)
[38017] Module object for pyimod01_archive is NULL!
Traceback (most recent call last):
  File "PyInstaller/loader/pyimod01_archive.py", line 27, in <module>
ModuleNotFoundError: No module named 'struct'
[38017] Module object for pyimod02_importers is NULL!
Traceback (most recent call last):
  File "PyInstaller/loader/pyimod02_importers.py", line 28, in <module>
ModuleNotFoundError: No module named 'pyimod01_archive'
Traceback (most recent call last):
  File "PyInstaller/loader/pyiboot01_bootstrap.py", line 17, in <module>
ModuleNotFoundError: No module named 'pyimod02_importers'
[38017] Failed to execute script 'pyiboot01_bootstrap' due to unhandled exception!
```

```
~/Downloads/hwi-2.2.1 ./hwi.py --debug enumerate
DEBUG:ledgercomm:=> b001000000
DEBUG:ledgercomm:<= 0107426974636f696e05322e312e310102 9000
DEBUG:ledgercomm:=> e105000100
DEBUG:ledgercomm:<= 0b675ab1 9000
WARNING:jade:No module named 'hwilib.devices.jadepy.jade_ble'
WARNING:jade:BLE scanning/connectivity will not be available
INFO:jade.tcp:Connecting to tcp:127.0.0.1:30121
DEBUG:root:Failed to connect to Jade simulator at tcp:127.0.0.1:30121
DEBUG:root:[Errno 61] Connection refused
[{"type": "ledger", "model": "ledger_nano_x", "label": null, "path": "DevSrvsID:429xxxxx31", "fingerprint": "0bxxxxb1", "needs_pin_sent": false, "needs_passphrase_sent": false}]


```

**Comment by giahuy98** [comment]
Not me but @d6n13l0l1v3r, he is using macOS 13.3.1
![image](https://user-images.githubusercontent.com/100193713/233124485-027cc695-e021-4c87-bc5b-20425b9f9eda.png)

**Comment by tongvanlinh** [comment]
@achow101 I got this problem in my mac mini also
![image](https://user-images.githubusercontent.com/12754994/233379039-5bf83caa-7e31-40b9-a296-6e9ed216b2ba.png)

**Comment by achow101** [comment]
@giahuy98 That is an unrelated issue that has been resolved for 2.3.0. It appears something broke in my macOS build environment.

@tongvanlinh 2.3.0 has support for Apple Silicon macs.
